src/overlays/actors/ovl_En_Bigslime/z_en_bigslime.c

```diff
@@ -54,289 +243,2911 @@ const ActorInit En_Bigslime_InitVars = {
+    Player* player;
+    s32 pad;
+
+    if (globalCtx->envCtx.unk_C3 == 3) {
+        globalCtx->envCtx.unk_C3 = 0xFF;
+    }
+
+    func_8019F540(0);
+    this->isAnimUpdate = SkelAnime_Update(&this->skelAnime);
+    if (this->actionFunc != EnBigslime_PlayCutscene) {
+        EnBigslime_ApplyDamageEffectGekko(this, globalCtx);
+    } else {
+        this->gekkoCollider.base.acFlags &= ~AC_HIT;
+    }
+
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc != EnBigslime_FormBigslime) {
+        Actor_SetVelocityAndMoveYRotationAndGravity(&this->actor);
+    } else {
+        Actor_SetVelocityAndMoveXYRotation(&this->actor);
+    }
+
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 20.0f, 40.0f, 80.0f, 0x1F);
+    this->gekkoCollider.dim.pos.x = (s16)this->actor.world.pos.x;
+    this->gekkoCollider.dim.pos.z = (s16)this->actor.world.pos.z;
+    if (this->gekkoCollider.base.acFlags & AC_ON) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->gekkoCollider.base);
+    }
+
+    if ((this->actor.update == EnBigslime_UpdateGekko) && (this->gekkoCollider.base.ocFlags1 & OC1_ON)) {
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->gekkoCollider.base);
+    }
+
+    EnBigslime_UpdateIceShardEffect(this);
+    if ((this->actionFunc != EnBigslime_StunGekko) &&
+        (this->gekkoCollider.dim.pos.y < (s16)(3.0f + GBT_ROOM_5_MIN_Y))) {
+        Vec3f vtxNorm;
+
+        if (((globalCtx->gameplayFrames % 4) == 0) || !isGekkoOnGround) {
+            player = GET_PLAYER(globalCtx);
+            vtxNorm.x = this->actor.world.pos.x;
+            vtxNorm.z = this->actor.world.pos.z;
+            vtxNorm.y = player->actor.world.pos.y + player->actor.depthInWater;
+            EffectSsGRipple_Spawn(globalCtx, &vtxNorm, 150, 550, 0);
+            isGekkoOnGround = true;
+        }
+    } else {
+        isGekkoOnGround = false;
+    }
+}
+
+void EnBigslime_SetSysMatrix(Vec3f* pos, GlobalContext* globalCtx, Gfx* dList, f32 scaleX, f32 scalez, f32 scaleY,
+                             s16 rotation, f32 alpha) {
+    f32 yDistMinY;
+    f32 xz;
+    MtxF* sysMatrix = Matrix_GetCurrentState();
+
+    yDistMinY = pos->y - scaleY - GBT_ROOM_5_MIN_Y;
+    yDistMinY = CLAMP((yDistMinY), 0.0f, (GBT_ROOM_5_CENTER_Y - GBT_ROOM_5_MIN_Y) / 2);
+    xz = 1.0f - (yDistMinY * (1.0f / 1550.0f));
+
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    POLY_OPA_DISP = Gfx_CallSetupDL(POLY_OPA_DISP, 0x2C);
+    sysMatrix->xx = xz;
+    sysMatrix->yy = 1.0f;
+    sysMatrix->zz = xz;
+    sysMatrix->xz = sysMatrix->xy = 0.0f;
+    sysMatrix->yz = sysMatrix->yx = 0.0f;
+    sysMatrix->zy = sysMatrix->zx = 0.0f;
+    sysMatrix->wx = pos->x;
+    sysMatrix->wy = GBT_ROOM_5_MIN_Y;
+    sysMatrix->wz = pos->z;
+    sysMatrix->xw = sysMatrix->yw = sysMatrix->zw = 0.0f;
+    sysMatrix->ww = 1.0f;
+
+    Matrix_RotateY(rotation, MTXMODE_APPLY);
+    Matrix_Scale(scaleX, 1.0f, scalez, MTXMODE_APPLY);
+    if ((u32)dList != (u32)&D_06011050) {
```

Comment:
Are the casts necessary?
```suggestion
    if (dList != D_06011050) {
```

Reply:
I get an error when I remove the `(u32)`, "Unacceptable operand of == or !=", so I suspect I'm messing up my `Gfx` vs `Gfx*` vs `Gfx**` vs `Gfx []` types. Doing `dList != &D_06011050[0]` matches just fine as well. Any suggestions?

For reference:
`Gfx* dList`
`Gfx D_06011050[]`